File: erlang/leap/leap.erl
-module(leap).
-export([leap_year/1]).

-spec leap_year(integer()) -> boolean().
leap_year(Y) when is_integer(Y) ->
    case Y rem 4 of
        0 -> case Y rem 100 of
                 0 -> Y rem 400 =:= 0;
                 _ -> true
             end;
        _ -> false
    end.

File: elixir/leap/leap.exs
defmodule Year do
  @doc """
  Returns whether 'year' is a leap year.

  A leap year occurs:

  on every year that is evenly divisible by 4
    except every year that is evenly divisible by 100
      except every year that is evenly divisible by 400.
  """
  @spec leap_year?(non_neg_integer) :: boolean
  def leap_year?(year) when rem(year, 4) === 0 do
    rem(year, 100) !== 0 or rem(year, 400) === 0
  end
  def leap_year?(_), do: false
end